src/pages/gyeongminLee/Login/Login.js

```diff
@@ -1,8 +1,151 @@
 import React, { Component } from 'react';
-
+import './Login.scss';
 class Login extends Component {
+  constructor() {
+    super();
+    this.state = {
+      idValue: '',
+      passwordValue: '',
+      // submitButtonDisabled: true,
+      // buttonColor: '#B2DFFC',
+    };
+  }
+
+  controlValue = event => {
+    if (event.target.type === 'text') {
+      // 해결함 !!!!!
+      //인풋이 여러개일때 이벤트 객체를 활용, 타입이 텍스트인 인풋을 찾는것
+      // <form id="login-form" onChange={this.controlValue()}> 라고 적혀있으면 컨트롤 밸류의 리턴값을 찾는데 없으므로 언디파인이나온다.
+      this.setState(
+        {
+          idValue: event.target.value,
+        },
+        () => this.controlActivation(event) //정리해야할것 - 기존의 컨트롤 액티베이션 함수를 태그에 넣을 필요없이
+      );
+
+      // 실행이 완료될때까지 기다리지 않는다 - setState -> 비동기 아이디 밸류 달라지기 전에 아래 함수 실행함
+    } else if (event.target.type === 'password') {
+      this.setState(
+        {
+          passwordValue: event.target.value,
+        },
+        () => this.controlActivation(event)
+      );
+    }
+  };
+
+  goToMain = e => {
+    fetch('http://10.58.0.86:8000/user/signin', {
+      method: 'post',
+      body: JSON.stringify({
+        email: this.state.idValue,
+        password: this.state.passwordValue,
+      }),
+    })
+      .then(response => response.json())
+      .then(
+        result => {
+          if (result.MESSAGE !== 'SUCCESS') {
+            //응답은 맞는경우도 있고 아닌경우도 있다 - if문으로 나누어 주어야된다
+            this.props.history.push('/main-Gyeongmin');
+            //alert(result.ACCESS_TOKEN);
+            localStorage.setItem('Token', result['ACCESS_TOKEN']);
+          }
+        }
+
+        //.then setItem('ACCESS_TOKEN', result.ACCESS_TOKEN)
+      );
+    // .then(localStorage.setItem('ACCESS_TOKEN',  JsonResponse.ACCESS_TOKEN));
+  };
+
   render() {
-    return <div>로그인</div>;
+    //console.log(JsonResponse['ACCESS_TOKEN']);
+    // const idValid = this.state.idValue.length >= 1; 멘토님의 것
+    //const inputValid =this.state.idValue.includes('@')&&this.state.password.length>=5
+    const idValid = this.state.idValue.includes('@');
+    const passwordValid = this.state.passwordValue.length >= 5;
```

Comment:
```suggestion
    const idValid = this.state.idValue.includes('@');
    const passwordValid = this.state.passwordValue.length >= 5;
    const isAllValueValid = idValid && passwordValid
    
                  backgroundColor={
              isAllValueValid   === true ? '#B2DFFC' : '#0095F6'
              }
              disabled={isAllValueValid === true ? false : true}
```
모두 유효한지에 대한 정보를 background, disabled에서 두번 사용하는 것 같은데 이런경우 변수화 해서 중복없애서 사용할 수 있겟네요!

Reply:
boolean은 true or false의 정보이기 때문에 변수명으로 is~~~로 사용하는 컨벤션이 있습니다!